blues/sparrow-reference-web-app
# Vhardy/cypresstests

# Problem Context

The Sparrow needs e2es to shore up its functionality.

Since our app has relatively limited functionality for the time being, I forwent multiple test files and I also forwent attempting to mock response data from Notehub (which takes a very long time to come back in the browser) because if all goes well we won't be depending on Notehub for data for that much longer. We should have a third party database sooner or later which will make data retrieval much quicker on the browser.

## Changes

Added a new e2e test file to test the app's general functionality and few screens, added a commands file for reusable commands to make test extension easier in the future, configured Cypress environment files to run locally versus in GH actions, set up GH actions to run on PRs that are open and ready for review.

## Screenshot (if applicable)

<img width="782" alt="Screen Shot 2022-02-09 at 2 45 28 PM" src="https://user-images.githubusercontent.com/20400845/153278655-4bbbda11-437a-40dc-87de-170062168e48.png">

## Testing

Unit / integration / e2e tests?

All e2es, but unit tests still pass too.

Steps to test manually?

1. If you'd like to run the tests manually, pull down the local branch.
2. Start up the local development server in one terminal
3. Type `yarn cypress:run` in another terminal to run the tests headless mode
4. Type `yarn cypress:open` in another terminal to run the tests in headed mode
5. See the tests pass

Any other sorts of testing notes to include?

## Any other related PRs

n/a

## Ticket(s)

https://www.pivotaltracker.com/story/show/181160046
